saleor/checkout/complete_checkout.py

```diff
@@ -542,17 +547,105 @@ def _process_payment(
                 customer_id=customer_id,
                 store_source=store_source,
                 additional_data=payment_data,
-                channel_slug=channel_slug,
+                channel_slug=checkout_info.channel.slug,
             )
         payment.refresh_from_db()
+        payment.complete_order = complete_order
```

Comment:
Agreed, but in that case it feels like `get_checkout_by_token` should select it with select_for_update. Webhooks already do that

Reply:
I added the lock in 1a67813. This way webhooks should no longer race condition with CheckoutComplete, e.g. for stripe we'll call `_process_payment_with_checkout` but `_get_checkout` should return None since another transaction should delete the checkout.

Reply:
This needs to be tested well. Especially with enabled webhooks, to make sure that we don't have any deadlocks.